### PR TITLE
Fix jnp.trapezoid on an empty integration axis

### PR DESCRIPTION
**Pull request by pulkit6732**
The fast path added for constant dx reads y[..., 0] and y[..., -1], which raises IndexError when the integration axis is empty. numpy returns 0.0 for this input, and the general path below already produces 0.0, so the fast path simply should not apply when the axis has no elements.

Adds empty-axis shapes to the existing test_trapezoid parametrisation.

### CODE REVIEW

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -6340,6 +6340,10 @@ def test_isdtype(self, dtype, kind):
         ((3, 10), (10,), 1.0, -1),
         ((3, 10), (3, 10), 1.0, -1),
         ((2, 3, 10), (3, 10), 1.0, -2),
+        # empty integration axis: the result is zero, as in numpy.
```
Comment (jakevdp) [inline tests/lax_numpy_test.py:6343]: This comment is unnecessary – all cases in the parameterization are compared to NumPy.

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -6521,8 +6521,9 @@ def trapezoid(y: ArrayLike, x: ArrayLike | None = None, dx: ArrayLike = 1.0,
     else:
       dx_array = moveaxis(diff(x_arr, axis=axis), axis, -1)
   y_arr = moveaxis(y_arr, axis, -1)
-  # Fast path: dx is constant along the integration axis.
-  if dx_array.ndim == 0 or dx_array.shape[-1] == 1:
+  # Fast path: dx is constant along the integration axis. It relies on the
+  # first and last elements existing, so it does not apply to an empty axis.
+  if y_arr.shape[-1] > 0 and (dx_array.ndim == 0 or dx_array.shape[-1] == 1):
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:6526]: I think this should be `if y_arr.shape[-1] > 1`, otherwise it's not really a fast path

**Review by jakevdp — CHANGES_REQUESTED** [review]
